src/main/java/io/vertx/core/file/impl/FileSystemImpl.java

```diff
@@ -1045,11 +1045,16 @@ public boolean accept(File dir, String name) {
             } else {
               files = file.listFiles(fnFilter);
             }
-            List<String> ret = new ArrayList<>(files.length);
-            for (File f : files) {
-              ret.add(f.getCanonicalPath());
+            if (null == files || files.length == 0) {
+              return new ArrayList<>();
+            } else {
+              List<String> ret = new ArrayList<>(files.length);
```

Comment:
If we make the above immutable, this list should be too, for consistency

Reply:
we cannot really make this immutable otherwise it will be a breaking change